### PR TITLE
add google-ad-titles answer

### PR DESCRIPTION
**Pull request by Mykhailo-Zub**

### CODE REVIEW

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+Output: 
+
+```
+[
+  'Oxylabs Premium Proxies - Residential Proxy Network',
+  'Bright Data #1 Data Platform - 7 Day-Free Trial - Join Now'
+]
+```
+
+Check [my code in the online IDE][3].
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md:85]:
```suggestion
```
Let's move it above, before the actual code.

Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md:85]:
```suggestion
```
Let's move it above, before the actual code.

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+```javascript
+const unirest = require("unirest");
+const cheerio = require("cheerio");
+
+function getData() {
+  return new Promise((resolve, reject) => {
+    const titles = [];
+    unirest
+      .get("https://www.google.com/search?q=oxylabs")
+      .headers({ Accept: "application/json", "Content-Type": "application/json" })
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md:12]:
```suggestion
      .headers({ "Accept": "application/json", "Content-Type": "application/json", "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36" })
```

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      // .proxy(proxy) //hided
+      .then((response) => {
+        const $ = cheerio.load(response.body);
+
+        $(".uEierd").each((i, el) => {
+          titles[i] = $(el).find(".ZINbbc div.v5yQqb a.cz3goc div.CCgQ5 span").text();
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md:18]:
```suggestion
          titles[i] = $(el).find(".v0nnCb span").text();
```

Changed to a shorter selector for Titles from Ad results.

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+Oxylabs - Premium Proxy - Best Paid Proxy Service 2022
+```
+
+Alternatively, you can use the [Google Ad Results API][1] from SerpApi. It solves captchas, rotates proxies, mimicking completely what a human will do. This guarantees that you get what users truly see. Check out the [Playground][2] for more.
+
+Usage:
+
+```javascript
+const SerpApi = require("google-search-results-nodejs");
+const search = new SerpApi.GoogleSearch("db088c5746dae73c45b3c1ce6cde6278af0e7aa38d2f0c0180db26174b5027dc");
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md:52]:
```suggestion
const mySecret = process.env['API_KEY']
const search = new SerpApi.GoogleSearch(mySecret);
```
Is it your API key? If so, please, regenerate it and update on replit as well: https://serpapi.com/manage-api-key ✨

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-cheerio.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const unirest = require("unirest");
+const cheerio = require("cheerio");
+
+function getData() {
+  return new Promise((resolve, reject) => {
+    const titles = [];
+    unirest
+      .get("https://www.google.com/search?q=oxylabs")
+      .headers({ Accept: "application/json", "Content-Type": "application/json" })
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-cheerio.js:9]:
```suggestion
      .headers({ "Accept": "application/json", "Content-Type": "application/json", "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.54 Safari/537.36" })
```

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-serpApi.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+const SerpApi = require("google-search-results-nodejs");
+const search = new SerpApi.GoogleSearch("db088c5746dae73c45b3c1ce6cde6278af0e7aa38d2f0c0180db26174b5027dc");
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-serpApi.js:2]:
```suggestion
const mySecret = process.env['API_KEY']
const search = new SerpApi.GoogleSearch(mySecret);
```

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-serpApi.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+const params = {
+  engine: "google",
+  q: "oxylabs",
+  location: "Austin, Texas, United States",
+  google_domain: "google.com",
+  gl: "us",
+  hl: "en",
+};
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-serpApi.js:11]:
```suggestion
const params = {
  engine: "google",                         // search engine 
  q: "oxylabs",                             // search query
  location: "Austin, Texas, United States", // location parameter
  google_domain: "google.com",              // google domain of the search
  gl: "us",                                 // contry of the search
  hl: "en"                                  // language of the search  
};
```

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-serpApi.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+const SerpApi = require("google-search-results-nodejs");
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-serpApi.js:1]:
```suggestion
// https://stackoverflow.com/questions/70853810/returned-null-response-from-unirest-block
// https://replit.com/@MikhailZub/Google-Ad-Results-with-NodeJS-SerpApi#index.js

const SerpApi = require("google-search-results-nodejs");
```

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-cheerio.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
+const unirest = require("unirest");
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-cheerio.js:1]:
```suggestion
// https://stackoverflow.com/questions/70853810/returned-null-response-from-unirest-block
// https://replit.com/@MikhailZub/Google-Ad-Results-with-NodeJS-SerpApi#index.js

const unirest = require("unirest");
```

Link to an answer and replit. This is helpful if you need to answer a similar question.

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/question.md [hunk 10]
```diff
@@ -0,0 +1 @@
+https://stackoverflow.com/questions/70853810/returned-null-response-from-unirest-block
```
Comment (dimitryzub) [inline answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/question.md:1]: Let's delete this file and add SO question to the [Cheerio](answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-cheerio.js)/[SerpApi ](answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/google-ad-titles-scrape-serpApi.js)solution answers. 

There's no need to create a separate file with a question.

**Review by dimitryzub — CHANGES_REQUESTED** [review]
@Mykhailo-Zub, thank you for the answer💯 

I've added a few changes and a few things I want to point out. Tag me whenever changes will be applied.

I'll go ahead and mark it as a rejected answer.

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md (hunk 4 above)
Comment (Mykhailo-Zub) [reply answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md]: Yes, it's my API key. Why i should regenerate it?

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+logData();
+```
+
+Output: 
+
+```
+Oxylabs - Premium Proxy - Best Paid Proxy Service 2022
+```
+
+Alternatively, you can use the [Google Ad Results API][1] from SerpApi. It solves captchas, rotates proxies, mimicking completely what a human will do. This guarantees that you get what users truly see. Check out the [Playground][2] for more.
```
[the remark this answers is not in the document]
Comment (dimitryzub) [reply answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md]: Makes sense 👍🏻 We need to change the wording in this part.

File: answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md (hunk 4 above)
Comment (dimitryzub) [reply answers/Mikhail/stackoverflow/google/google-search/google-ad/google-add-titles/answer.md]: > Yes, it's my API key. Why I should regenerate it?

1. Users that don't have a SerpApi account can use your API directly without registering an account i.e not paying for it.
2. This is a potential vulnerability for hackers to access SerpApi. 

Please, regenerate it, and update the [replit repl](https://replit.com/@MikhailZub/Google-Ad-Results-with-NodeJS-SerpApi#index.js) as well.

**Review by dimitryzub — APPROVED** [review]
@Mykhailo-Zub That's great, thank you for the answer 🐳

You can go ahead and publish the answer to Stackoverflow. Once published, please, comment a link to your answer in this PR. [Here's an example from a different RP](https://github.com/dimitryzub/stackoverflow-answers-archive/pull/7#issuecomment-1105165128).

**Comment by Mykhailo-Zub** [comment]
https://stackoverflow.com/questions/70853810/returned-null-response-from-unirest-block/72199294#72199294

**Pull request merged**